bookstore/clone.py

```diff
@@ -40,6 +37,68 @@ class BookstoreCloneHandler(IPythonHandler):
     `Jupyter Notebook reference on Custom Handlers <https://jupyter-notebook.readthedocs.io/en/stable/extending/handlers.html#registering-custom-handlers>`_
     """
 
+    def initialize(self):
+        """Helper to retrieve bookstore setting for the session."""
+        self.bookstore_settings = BookstoreSettings(config=self.config)
+
+    @web.authenticated
+    async def get(self):
+        """GET /bookstore/clone?s3_bucket=<your_s3_bucket>&s3_key=<your_s3_key>
+
+        Renders an options page that will allow you to clone a notebook
+        from a specific bucket via the Bookstore cloning API.
+        """
```

Comment:
Let's add a note in this docstring about the distinction between s3_object_key and s3_key. In particular let's clarify which that the key passed as a parameter in the example above is the s3_key (not the s3_object_key)

Reply:
I think this comment might be coming from the same confusion as earlier. 
The `s3_key` in the above as a parameter is the `s3_object_key` internal to the function. 

```python
        s3_object_key = self.get_argument("s3_key")
        if s3_object_key == '' or s3_object_key == '/':
            raise web.HTTPError(400, "Must have a key to clone from")
```

This is an attempt to fit with convention that most S3 endpoints & clients have where what is provided is only the "Bucket" and the "Key", and no authentication or authorization ever happens over the web. In those contexts
`Key`, `s3Key`, `s3_key`, `S3Key` is the parameter that we landed on calling `s3_object_key` internal to the function. 

The stuff around authentication and authorization are usually passed as command line args, via configuration files, or environment variables (from https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-envvars.html)
AWS_ACCESS_KEY_ID – Specifies an AWS access key associated with an IAM user or role. AWS_SECRET_ACCESS_KEY – Specifies the secret key associated with the access key. This is essentially the "password" for the access key.

I think these might be unsafe to pass via URL, which it's possible that so much consolidation around the meaning of `Key` in endpoints/clients being the "object key".

If you think we should change the URL parameter to `s3_object_key` I'm on board… as it would be clearer for a user who is unfamiliar with s3 clients. We will then be deviating further from the norm in how most s3 clients & endpoints backed by s3 clients. And so we should be aware we are making that tradeoff.

Reply:
It's fine to leave as is. I agree re: passing around credentials.

I guess I am just trying to determine if there is any difference between the s3_key in the example GET request and the s3_object_key which is used internally by the method.

Reply:
There isn't… but we have the `s3_key` function in the `s3_paths` module, which is why we have the comment above the declaration. Is there any other way you would suggest conveying that?

```
        # s3_paths module has an s3_key function; s3_object_key avoids confusion
        s3_object_key = self.get_argument("s3_key")
```